fdbserver/KmsConnectorInterface.h

```diff
@@ -129,19 +152,40 @@ struct KmsConnLookupEKsByDomainIdsRep {
 	}
 };
 
+struct KmsConnLookupDomainIdsReqInfo {
+	constexpr static FileIdentifier file_identifier = 8980149;
+	EncryptCipherDomainId domainId;
+	EncryptCipherDomainName domainName;
+
+	KmsConnLookupDomainIdsReqInfo() : domainId(ENCRYPT_INVALID_DOMAIN_ID) {}
+	explicit KmsConnLookupDomainIdsReqInfo(const EncryptCipherDomainId dId, StringRef name, Arena& arena)
+	  : domainId(dId), domainName(StringRef(arena, name)) {}
+
+	bool operator==(const KmsConnLookupDomainIdsReqInfo& info) const {
+		return domainId == info.domainId && (domainName.compare(info.domainName) == 0);
+	}
+
+	template <class Ar>
+	void serialize(Ar& ar) {
+		serializer(ar, domainId, domainName);
+	}
+};
+
 struct KmsConnLookupEKsByDomainIdsReq {
 	constexpr static FileIdentifier file_identifier = 9918682;
-	std::vector<EncryptCipherDomainId> encryptDomainIds;
+	Arena arena;
+	std::vector<KmsConnLookupDomainIdsReqInfo> encryptDomainInfos;
 	Optional<UID> debugId;
 	ReplyPromise<KmsConnLookupEKsByDomainIdsRep> reply;
 
 	KmsConnLookupEKsByDomainIdsReq() {}
-	explicit KmsConnLookupEKsByDomainIdsReq(const std::vector<EncryptCipherDomainId>& ids, Optional<UID> dbgId)
-	  : encryptDomainIds(ids), debugId(dbgId) {}
+	explicit KmsConnLookupEKsByDomainIdsReq(const std::vector<KmsConnLookupDomainIdsReqInfo>& infos,
+	                                        Optional<UID> dbgId)
+	  : encryptDomainInfos(infos), debugId(dbgId) {}
 
 	template <class Ar>
 	void serialize(Ar& ar) {
-		serializer(ar, encryptDomainIds, debugId, reply);
+		serializer(ar, arena, encryptDomainInfos, debugId, reply);
```

Comment:
Given `Arena` is one of the member of the struct, it needs to be serialized as well. The intention is given `KmsConnLookupDomainIdsReqInfo` stores a StringRef, supply an arena to avoid reference going out of scope.